jax/experimental/export/shape_poly.py

```diff
@@ -242,7 +242,12 @@ def evaluate(self, env: DimVarEnv):
       elif self.operation == _DimAtom.MOD:
         return divmod(*operand_values)[1]  # type: ignore
       elif self.operation == _DimAtom.NON_NEGATIVE:
-        return lax.max(operand_values[0], 0)
+        operand = operand_values[0]
+        if core.is_symbolic_dim(operand):
+          return core.non_negative_dim(operand)
+        if core.is_constant_dim(operand):
+          return max(operand, 0)
+        return lax.max(operand, 0)
```

Comment:
When is this branch taken? I thought a dimension is either symbolic or constant, i.e. there are no other options.

Reply:
This function is also used to evaluate symbolic expressions in contexts when the dimension variables are mapped to JAX tracers (e.g., when we evaluate shapes for lowering) or TF ops (when we evaluate shapes for jax2tf lowering). In those later cases we want JAX primitives to get involved but not in other cases.